packages/core/src/fusion/Filter.js

```diff
@@ -21,6 +22,10 @@ export default class Filter {
    * @returns {Filter} filtered source
    */
   static create (p, source) {
+    if (isCanonicalEmpty(source)) {
+      return source
```

Comment:
Quick q: since we'll likely need to return `empty()` in the majority of other cases, should we return it here just for the sake of source code consistency?  (In terms of types and functionality, there's no difference)

Reply:
I was actually micro-optimizing here. Your call.

Reply:
I'm cool with it as is.